components/LocationInput.tsx

```diff
@@ -8,126 +8,164 @@ import React, {
+        placeholder: 'Search a location',
         marker: false,
         reverseGeocode: true,
       });
-      if (geocoderContainer.current) {
-        geocoderContainer.current.appendChild(geocoder.onAdd(map.current));
+
+      if (!canEdit) {
+        // Prevent user from changing the geocoder input when the map isn't editable
+        geocoder.setBbox([0, 0, 0, 0]);
       }
 
+      geocoderContainer.current?.appendChild(geocoder.onAdd(map.current));
+
       const marker = new mapboxgl.Marker({
         draggable: false,
         color: '#FFD053',
-      }).setLngLat([lng, lat]);
+      }).setLngLat([originalLat, originalLng]);
 
-      map.current.on('load', () => setIsMapReady(true));
+      map.current.on('load', () => {
+        if (shouldZoomIntoOriginal) {
+          geocoder.query(`${originalLat}, ${originalLng}`).setFlyTo(canEdit);
+        }
+      });
 
       map.current.on('idle', () => {
         // Enables fly to animation on search
         geocoder.setFlyTo(true);
       });
 
-      map.current.on('movestart', () => {
-        setIsSubmissionEnabled(true);
-      });
-
-      let savedLat = originalLng;
-      let savedLng = originalLat;
+      let savedLat = originalLat;
+      let savedLng = originalLng;
 
       const moveHandler = () => {
         const newLng = map.current!.getCenter().lng;
         const newLat = map.current!.getCenter().lat;
 
-        setLng(newLng);
-        setLat(newLat);
-        setLngExternal(newLng);
-        setLatExternal(newLat);
+        setMapCoordinates({ longitude: newLng, latitude: newLat });
 
         savedLat = newLat;
         savedLng = newLng;
+        updateMarker(
+          marker,
+          map.current!,
+          zoomLevelThreshold,
+          newLng,
+          newLat,
+          canEdit
+        );
 
         setZoom(map.current!.getZoom());
-        updateMarker(marker, map.current!, zoomLevelThreshold, newLng, newLat);
       };
 
-      map.current.on('move', moveHandler);
+      map.current.on('movestart', () => {
+        if (popup) {
+          popup.remove();
+        }
+      });
 
       map.current.on('moveend', () => {
-        const isPastZoomThreshold = map.current!.getZoom() > zoomLevelThreshold;
+        const isPastZoomThreshold =
+          map.current!.getZoom() >= zoomLevelThreshold;
         if (isPastZoomThreshold) {
+          moveHandler();
+
           // update the search box location based on the final latitude/longitude
           geocoder.query(`${savedLat}, ${savedLng}`).setFlyTo(false);
+
+          if (popup === null && map.current && canEdit) {
+            popup = new mapboxgl.Popup({
+              offset: [0, 10],
+              anchor: 'top',
+              closeOnClick: false,
+              closeButton: false,
+              className: 'site-map',
+            })
+              .setLngLat([savedLng, savedLat])
+              .setHTML('<p>Drag map to pinpoint exact location.</p>')
```

Comment:
Could we make this change to the "Drag map to pinpoint exact location" step be triggered by clicking the "next" button (which can perhaps be enabled after we're past the requisite zoom threshold) rather than having the transition be automatic?

Reply:
I don't understand why we'd need to do this from a user experience perspective. Wouldn't this be an unnecessary, unintuitive middle step? Can't we combine entering your address and making sure the pin is in the right spot into one step?

Reply:
Why this could hurt the user experience:

- If the user needs to change their address, they'd need to go back 2 steps instead of 1.
- Most users won't need to adjust the pin since the solar panel will be on their house. Hence, we have a "optional" second step in the middle of the form.

Reply:
Another concern: users may need to enter multiple addresses before confirming the location and moving the pin. However, how likely is this to happen? I argue that most homeowners will enter the correct address that first time, and so automatically moving to the pinning step is fine.

Reply:
> Why this could hurt the user experience:
> 
> * If the user needs to change their address, they'd need to go back 2 steps instead of 1.

I think this is pretty valid, but to address this, we could make it so that "back" on the metadata input form goes back to the first step, rather than the "pinpoint location" step in the middle

Reply:
A bigger issue that might push us toward having that intermediary step rather than automatically moving forwards based on zoom value is that, if a user decides to input a location by manipulating the map, the transition to having a location pin show up is quite abrupt, and might be confusing, as the location pin appears when you're still quite far out from being able to pinpoint any exact location.

I feel like having the location pin appear immediately after you surpass some zoom threshold might be a bit confusing, and it simply doesn't make the application look well-polished or smooth. Furthermore, since this isn't an actual step, there is no "back" functionality—after the location pin appears, there's no way to revert back to the original state ("where is your solar panel located")

https://user-images.githubusercontent.com/62641231/227739333-ed45d377-e4c1-41d9-b904-bec16f62a4ec.mov

Reply:
Also, to add another point—one of the main preliminary reasons that led to us even considering a redesign in the first place was to avoid having the "precise location" pin pop up based on threshold value in the first place.